### DISCUSSION TITLE
Whats wrong with having a valid session when creating a new account?

### DISCUSSION THREAD

**Discussion by MJEND7** [ory/kratos#3076] [Q&A]
I was playing around with Kratos, and was wondering, when you have a session and you want to create a new account, it will not let you, was really just wondering if there was any particular reason for that or if it's just something that exists since in most cause it's not wanted.

There are a few use cases I can see when you could want that mainly if you have a system were uses are allowed to have multiple accounts and want them linked straight of that bat, like just putting the id of the current identity into the new one created. 

It could also relate to this issue https://github.com/ory/kratos/issues/765 which is a use case for more then one *type of account* which you could want to login and out of and or create related accounts, like an admin account that relates to my current account I am creating it from. (Which is not hard to implement if you could have a valid session on creation of another account)

Answers are greatly appreciated😄!

**Comment by vinckr** [comment]
Hello @MJEND7 

> it's just something that exists since in most cause it's not wanted.

I am not sure if there is a security reason, but IMO this sounds most reasonable. Most want their users to only have one account at a time. 
If you are looking for something like "roles" / RBAC, then I suggest you check out [Ory Keto](https://www.ory.sh/docs/keto/)
There is also [account linking](https://www.ory.sh/docs/kratos/social-signin/link-multiple-provider-account) but AFAIK that is for the same (email) identifier, so if you have multiple options to authenticate users can link the other options to their one account.

If you could share some more detailson your use case, that would also be appreciated!

**Reply by MJENDIDS** [reply]
Hey sorry for the super late replay, issue with this answer is that when I send the sign up request there can be 2 types of accounts in my case. Think of it like a main account and a sub account you can auth into to, they also has a different schemas. Currently I have built my own system to handle more then one schema with Ory as the base but I don't think it is the correct way of doing things. Could keto provide a solution?
